js/app/Visualization/UI/GenerateReportDialog.js

```diff
@@ -179,25 +129,33 @@ define([
 
     startup: function() {
       var self = this;
-      self.inherited(arguments);
 
+      self.inherited(arguments);
       self.addChild(self.reportDialog);
     },
 
     handleAccept: function() {
-      var url = this.reportDialog.getReportUrl();
+      var self = this;
+
+      var url = self.getReportUrl();
       var contentTypeHeaders = { "Content-Type": "application/json;charset=UTF-8" };
-      var additionalHeaders = this.report.datamanager.headers;
+      var additionalHeaders = self.report.datamanager.headers;
       var headers = _.extend(contentTypeHeaders, additionalHeaders);
-      var body = JSON.stringify(this.reportDialog.getReportData());
+      var body = JSON.stringify(self.reportDialog.getReportData());
       Ajax.post(url, headers, body, function(err, result) {
         SimpleMessageDialog.show("Report generation", result.message);
       });
-      this.hide();
+      self.hide();
     },
 
     handleCancel: function() {
       this.hide();
+    },
+
+    getReportUrl: function() {
+      var self = this;
+
+      return "" + self.report.reportableAnimation.args.source.args.url + "/reports";
```

Comment:
This is to force javascript's horrible type system to treat `self.report.reportableAnimation.args.source.args.url` as a string. Weird things can happen if you don't do it. ES6 luckily has interpolated strings, so this is not necessary... but we don't get support for ES6 with dojo... so... we are stuck with this.